src/internet/spotifywebapi/spotifywebapiservice.cpp

```diff
@@ -0,0 +1,176 @@
+/* This file is part of Clementine.
+   Copyright 2021, Kenman Tsang <kentsangkm@pm.me>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "spotifywebapiservice.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QtDebug>
+
+#include "3rdparty/qtiocompressor/qtiocompressor.h"
+#include "core/application.h"
+#include "core/network.h"
+#include "globalsearch/globalsearch.h"
+#include "globalsearch/spotifywebapisearchprovider.h"
+#include "ui/iconloader.h"
+
+static constexpr auto kServiceName = "SpotifyWebApi";
+
+SpotifyWebApiService::SpotifyWebApiService(Application* app,
+                                           InternetModel* parent)
+    : InternetService(kServiceName, app, parent, parent),
+      network_(new NetworkAccessManager{this}) {
+  app_->global_search()->AddProvider(
+      new SpotifyWebApiSearchProvider(app_, this));
+}
+
+SpotifyWebApiService::~SpotifyWebApiService() {}
+
+QStandardItem* SpotifyWebApiService::CreateRootItem() {
+  root_ = new QStandardItem(IconLoader::Load("spotify", IconLoader::Provider),
+                            kServiceName);
+  return root_;
+}
+
+void SpotifyWebApiService::LazyPopulate(QStandardItem* item) {}
+
+void SpotifyWebApiService::Search(int searchId, QString queryStr) {
+  if (std::chrono::system_clock::now().time_since_epoch() >= tokenExpiration_) {
+    QNetworkRequest request{
+        QUrl{"https://open.spotify.com/"
+             "get_access_token?reason=transport&productType=web_player"}};
+    request.setRawHeader("Accept-Encoding", "gzip");
+
+    QNetworkReply* reply = network_->get(request);
+    connect(reply, &QNetworkReply::finished, [=]() {
+      reply->deleteLater();
+
+      OnTokenReady(ParseJsonReplyWithGzip(reply), searchId, queryStr);
+    });
+  } else {
+    OnReadyToSearch(searchId, queryStr);
+  }
+}
+
+void SpotifyWebApiService::OnTokenReady(const QJsonDocument& json, int searchId,
+                                        QString queryStr) {
+  if (!json.isEmpty()) {
+    token_ = json["accessToken"].toString();
```

Comment:
This operator method was added in Qt 5.10, but Debian Stretch is still at 5.7. I think you can use json.object() to get the QJsonObject and get the values from that.

Reply:
Thx. This issue is resolved